mobile/library/common/jni/android_jni_utility.cc

```diff
@@ -17,15 +17,15 @@
 bool is_cleartext_permitted(absl::string_view hostname) {
 #if defined(__ANDROID_API__)
   envoy_data host = Envoy::Data::Utility::copyToBridgeData(hostname);
-  JNIEnv* env = Envoy::JNI::get_env();
-  jstring java_host = Envoy::JNI::native_data_to_string(env, host);
+  Envoy::JNI::JniHelper jni_helper(Envoy::JNI::get_env());
```

Comment:
Would it make sense to have JniHelper call `Envoy::JNI::get_env()` in the constructor to simplify usage?

Reply:
Unfortunately the use of `Envoy::JNI::get_env()` is not consistent. Some will use it and some won't. I believe `Envoy::JNI::get_env()` is also very Android specific, which means it won't work when running on HotSpot VM (non Android). In the future PR, I plan on fixing this as well, which could possibly be using `Envoy::JNI::get_env()` everywhere (or removing it).

Reply:
Oh, I see. How confusing! Thanks for working on this.